### DISCUSSION TITLE
How to access custom endpoint from another custom endpoint/hook?

### DISCUSSION THREAD

**Discussion by koljam** [directus/directus#7991] [Archive]
I made a custom endpoint for my own mailing system. Now I want to use that endpoint in other endpoints or hooks. Is there another possibility than making an http-request from the server to itself?

The schema works like this
- POST /mailer Endpoint -> receives data, retrieves mail-template from collection, replaces placeholders with given data and send E-Mail
- users.create hook -> sends data to /mailer to send welcome mail
- users.delete hook -> sends data to /mailer to send byebye mail
- ...

Thanks in advance!

**Comment by koljam** [comment]
I have gone a different route but I will let this question stay here in case somebody gets the same problem.
Look at the following discussion:
#7309 
I made a .js file in another folder which I can now `const Mailer = require("../path/to/jsfile.js")` and inside I have  `module.exports = async function Mailer(...`
This works for my usecase and is probably the way to go: require shared functions from another file and don't put them directly into the endpoint. If you have suggestions to make this better, please let me know!

**Reply by rijkvanzanten** [reply]
> This works for my usecase and is probably the way to go: require shared functions from another file and don't put them directly into the endpoint. If you have suggestions to make this better, please let me know!

This is exactly how I would do it 🙂
